### ISSUE TITLE
parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"

### ISSUE THREAD

**Issue by ricemeisterio** [hashicorp/terraform-provider-azurerm#20240]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.41.0

### Affected Resource(s)/Data Source(s)

azurerm_log_analytics_workspace

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.41.0" # azurerm_log_analytics_workspace originally created with version "3.7.0", when upgrading to 3.41.0 the Error appears
    }
  }

  backend "azurerm" {}
}
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_log_analytics_workspace" "example" {
  name                = "acctest-01"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "PerGB2018"
  retention_in_days   = 30
}
```

### Debug Output/Panic Output

```shell
│ Error: parsing "/subscriptions/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa/resourcegroups/example-resources/providers/microsoft.operationalinsights/workspaces/acctest-01": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
│
│   with azurerm_log_analytics_workspace.example,
│   on Stamp\azurerm_log_analytics_workspace.tf line 1, in resource "azurerm_log_analytics_workspace" "example":
│    1: resource "azurerm_log_analytics_workspace" "example" {
│
╵
```

### Expected Behaviour

Resource plans and applies successfully similar to 3.7.0

### Actual Behaviour

│ Error: parsing "/subscriptions/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa/resourcegroups/example-resources/providers/microsoft.operationalinsights/workspaces/acctest-01": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
│
│   with azurerm_log_analytics_workspace.example,
│   on Stamp\azurerm_log_analytics_workspace.tf line 1, in resource "azurerm_log_analytics_workspace" "example":
│    1: resource "azurerm_log_analytics_workspace" "example" {
│
╵

### Steps to Reproduce

# Create azurerm_log_analytics_workspace with AzureRM Provider Version "3.7.0"
# Update to  AzureRM Provider Version "3.41.0"
terraform init -upgrade
terraform plan

# Upgraded from 3.7.0 to 3.41.0 and the deployment of azurerm_log_analytics_workspace is now requiring title case for 'resourceGroups' in the id

Our configuration has the resource ID in lowercase, 'resourcegroups'.

**Comment by myc2h6o** [comment]
Hi @ricemeisterio thanks for opening the issue! This is usually due to a resource being imported in previous version with incorrect casing, and fails the validation of the new sdk which is stricter, the update to the new sdk is introduced in v3.19.0 https://github.com/hashicorp/terraform-provider-azurerm/commit/27c44ab7ced28f73baca7da18ef102c5ca28fbc3#diff-d775cce03ba9847dcc2154dd122d37569398867d8674508428558dd900e7c6d5R283. This resource is missing a state migration function during the update, we'll fix the issue by adding it.

**Comment by myc2h6o** [comment]
Though not recommended to modify the state file manually, as a temporary mitigation, the id in the state file could be changed to the expected casing before the fix is done.

**Comment by ricemeisterio** [comment]
Hi @myc2h6o - many thanks for your update. Yes, you are correct about the import. I had forgotten this.

I have decided to modify the state file manually, as a temporary mitigation, and this has been successful. 

I had to address segment "resourcegroups" to be "resourceGroups"
and on the second plan, also address the segment below, adding here in case someone is searching in the future.
parsing segment "staticMicrosoftOperationalInsights": expected the segment "microsoft.operationalinsights" to be "Microsoft.OperationalInsights".

Updating the ids for both of these segments allowed me to plan and apply.

I see a long-term fix is in review - great to have such a quick response.
